### DISCUSSION TITLE
Configuring integration: Time Out connecting to device

### DISCUSSION THREAD

**Discussion by ARTSRAY** [leeyuentuen/alfen_wallbox#87] [Q&A] [answered]
My Eneco Alfen Eve Pro is on my IOT network. I granted access to the IOT VLAN for HA. I can ping the Eneco charger from HA (HAOS is on a Hyper-V VM). When trying to configure the integration after installing it through HACS I receive a time out when trying to connect to the charger (during setup of the integration).

What could be wrong? I am able to connect to other devices using integrations, this one fails.

Thank you.

**Comment by leeyuentuen** [comment]
maybe provide log?

could it be that there is myeve app opened? normally alfen device support only one connection.

**Reply by ARTSRAY** [reply]
Thanks for your swift reply, no other connection, I also rebooted the charger to assure no other connections were made.
About the log, I am just starting to learn HA. Can you tell me where to find the log? I will post it here.

**Reply by leeyuentuen** [reply]
there are 2 locations for the log,
one is from the system: 
`Settings -> system-> logs`


the other is from the integration self:  
`settings -> device and service -> alfen wallbox -> (on the left) enable debug logging.`
after you are done with adding the integration (or error), stop the "enable debug logging" and it will download the log file

**Reply by ARTSRAY** [reply]
It is not (yet) listed within my device and services panel, when trying to Add the Integration, the procedure cannot finish because there is a time out to the Eneco Wallbox.

This is the system log:
Logger: homeassistant.loader
Source: loader.py:581
First occurred: December 17, 2023 at 2:59:09 PM (2 occurrences)
Last logged: December 17, 2023 at 2:59:09 PM

We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
We found a custom integration alfen_wallbox which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant


I also see this one, if this is the IP address for the Wallbox then this is incorrect. This IP number (.177) does not exist.

Logger: pymodbus.logging
Source: components/modbus/modbus.py:436
First occurred: December 17, 2023 at 12:29:33 PM (19666 occurrences)
Last logged: 11:28:42 AM

Connection to (192.168.20.177, 502) failed: [Errno 113] Host is unreachable
Connection to (192.168.20.177, 502) failed: timed out

**Reply by leeyuentuen** [reply]
if you are using myeve app (mobile) or ace installer (windows) could you reach that wallbox? it seems this look like more an network issue (firewall)?. 

a wait, you are using something with modbus? normally this integration is using polling instead of modbus

**Reply by leeyuentuen** [reply]
when you add them, it should look like this:

![image](https://github.com/leeyuentuen/alfen_wallbox/assets/1487966/43b223b3-586f-440a-9405-978abb8eb119)

**Comment by ARTSRAY** [comment]
Yes I am using modbus connection which I enabled within the Alfen Eve Pro charger. Do you mean that modbus connection is not supported?

**Reply by leeyuentuen** [reply, marked as answer]
this integration is not using modbus
